### ISSUE TITLE
SDPA silently returns wrong results when q/k/v are non-contiguous views

### ISSUE THREAD

**Issue by nikheal2** [torch-spyre/torch-spyre#4331]
### What happened?

Calling F.scaled_dot_product_attention on the Spyre device returns numerically wrong output when the query/key/value tensors are non-contiguous views instead of contiguous tensors.

The inputs themselves are correct, they match a CPU computation to fp16 tolerance (~0.004). Only the attention output is wrong, by a max absolute difference of ~1.0 to 1.7 against the same call on CPU.

There is no error, no exception, and no FallbackWarning. The operation runs on the card and produces confident garbage.

Workaround is to force the operands into fresh contiguous memory:

q = q.contiguous() before the call
F.pad(q, ...) before the call (padding for any reason, then cropping after)
Nothing else about the call changes — same values, same shapes, same dtype, same mask.

How we found it. 
A vision-attention test failed only at patch counts that were already a multiple of 64, and passed at every other count. That looked like a sequence-length bug for a while. It wasn't: the non-aligned path happened to call F.pad for stick alignment, and that padding was quietly repairing the layout. Once the length was held fixed and only contiguity varied, the pattern was unambiguous.

### What did you expect to happen?

Attention output identical to CPU within fp16 tolerance, regardless of whether the operands are contiguous.

If a layout genuinely can't be supported, the operation should raise or emit a FallbackWarning — not return wrong numbers silently

### How can we reproduce it?

Use below script:

```
import torch
import torch.nn.functional as F
import torch_spyre  # noqa: F401

B, H, L, D = 1, 4, 64, 64
DTYPE, SCALE = torch.float16, D**-0.5
dev = torch.device("spyre")

torch.manual_seed(0)
x = torch.randn(B, L, H * D, dtype=DTYPE)
w = torch.randn(H * D, 3 * H * D, dtype=DTYPE) * 0.02

keep = torch.ones(L, L, dtype=torch.bool).tril()
mask = torch.zeros(L, L, dtype=DTYPE).masked_fill(~keep, torch.finfo(DTYPE).min)[None, None]


def build(device):
    """q/k/v the way any attention layer builds them: chunk + reshape + transpose."""
    qkv = torch.matmul(x.to(device), w.to(device))
    return [p.reshape(B, L, H, D).transpose(1, 2) for p in qkv.chunk(3, dim=-1)]


q_ref, k_ref, v_ref = build(torch.device("cpu"))
expected = F.scaled_dot_product_attention(q_ref, k_ref, v_ref, attn_mask=mask, scale=SCALE)

q, k, v = build(dev)
m = mask.to(dev)

# Inputs are correct on the card.
print("q diff      ", (q.cpu().float() - q_ref.float()).abs().max().item())

bad = F.scaled_dot_product_attention(q, k, v, attn_mask=m, scale=SCALE)
print("out diff    ", (bad.cpu().float() - expected.float()).abs().max().item())

good = F.scaled_dot_product_attention(
    q.contiguous(), k.contiguous(), v.contiguous(), attn_mask=m, scale=SCALE
)
print("out+contig  ", (good.cpu().float() - expected.float()).abs().max().item())

```

### Any environmental details we need to know?

torch-spyre: 5e715ea70c4aa9e76e45b7fbae6b92a6d7658764
torch: 2.13.0a0+gitcf30153
Python 3.12, Linux x86_64
dtype float16 throughout
Shapes: [1, 4, 64, 64] and [1, 4, 128, 64], head_dim 64, additive fp16 causal mask [1, 1, L, L]

### Anything else we need to know?

I din't face this issue on this specific torch-spyre commit e02b78ba35f9a1a69a458c3149e9c01d9f4fa6a8 , so maybe this is regression.

### Relevant log output

```Shell
Stage-by-stage diff of that attention forward, CPU vs card, showing the inputs clean and the divergence appearing exactly at SDPA:



     stage                  shape   max abs diff
  qkv_proj           (1, 64, 768)       0.004883
    rope_q         (1, 64, 4, 64)       0.003906
    rope_k         (1, 64, 4, 64)       0.004395
      sdpa         (1, 4, 64, 64)       1.282227  <-- diverges
   reshape           (1, 64, 256)       1.282227
    o_proj           (1, 64, 256)       0.448486
```

**Comment by ani300** [comment]
https://github.com/torch-spyre/torch-spyre/pull/4299 fixes this issue and was merged before the weekend
